src/couch/src/couch_debug.erl

```diff
@@ -794,6 +795,68 @@ id("couch_file:init" ++ _, Pid, _Props) ->
 id(_IdStr, _Pid, _Props) ->
     "".
 
+first_call(Pid) ->
+    IC =
+        case process_info(Pid, initial_call) of
+            {initial_call, IC0} -> IC0;
+            undefined -> undefined
+        end,
+    Dict =
+        case process_info(Pid, dictionary) of
+            {dictionary, Dict0} -> Dict0;
+            undefined -> []
+        end,
+    MaybeCall = proplists:get_value('$initial_call', Dict, IC),
+    proplists:get_value(initial_call, Dict, MaybeCall).
+
+find_by_first_call(Module, Function) ->
+    FilterPids = fun(Pid) ->
+        case first_call(Pid) of
+            {Module, Function, _} -> true;
+            _ -> false
+        end
+    end,
+    lists:filter(FilterPids, processes()).
+
+view_report() ->
+    IndexerPids = find_by_first_call(couch_index, init),
+    Pid = lists:nth(1, IndexerPids),
+    {ok, IdxState} = couch_index:get_state(Pid, 0),
+    case element(1, IdxState) of
```

Comment:
The `#mrst` record should be available once you move it into `couch_mrview_debug.erl`

```
case IdxState of
    #mrst{} ->
```